components/org.wso2.carbon.identity.user.ekyc/src/main/java/org/wso2/carbon/identity/user/ekyc/UserEKYCConnectorImpl.java

```diff
@@ -0,0 +1,182 @@
+            UserEKYCException, ConfigurationManagementException, UserStoreException {
+        EKYCVerifiedCredentialDTO ekycVerifiedCredentialDTO = EKYCVerifiedCredentialDAO
+                .getInstance().getUserEKYCVC(sessionId, userId, tenantId);
+        if (isUpdatePossible(ekycVerifiedCredentialDTO, tenantId)) {
+            EKYCVerifiedCredentialDataDTO ekycVerifiedCredentialDataDTO = getEkycVerifiedCredentialDataDTO
+                    (ekycVerifiedCredentialDTO);
+            Map<String, String> claimsMapping = getClaimsMapping();
+            Map<String, String> mappedClaims = ekycVerifiedCredentialDataDTO.getClaims().entrySet().stream()
+                    .filter((entry) -> claimsMapping.keySet().contains(entry.getKey()))
+                    .map((entry) -> getMappedClaim(entry, claimsMapping))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            log.debug("Updating user claims " + StringUtils.join(mappedClaims));
+            getUserStoreManager(tenantId).setUserClaimValuesWithID(userId, mappedClaims, null);
+        }
+    }
+
+    private IDVService getIdvService() {
+        return EKYCServiceDataHolder.getInstance().getIdvService();
+    }
+
+    private EKYCVerifiedCredentialDataDTO getEkycVerifiedCredentialDataDTO(EKYCVerifiedCredentialDTO
+                                                                                   ekycVerifiedCredentialDTO) {
+        return new Gson().fromJson(ekycVerifiedCredentialDTO
+                .getVerifiedCredential(), EKYCVerifiedCredentialDataDTO.class);
+    }
+
+    private Map<String, String> getClaimsMapping() throws ConfigurationManagementException, UserEKYCException {
+        Map<String, String> claimsMapping = getEKYCConfiguration().getClaimsMapping();
+        if (claimsMapping == null) {
+            throw new UserEKYCException(UserEKYCConstants.ErrorMessages.CLAIMS_MAPPING_MISSING_ERROR);
+        }
+        return claimsMapping;
+    }
+
+    private Map.Entry<String, String> getMappedClaim(Map.Entry<String, Object> claim, Map<String, String>
+            claimsMapping) {
+        return new AbstractMap.SimpleEntry<String, String>(
+                claimsMapping.get(claim.getKey()),
+                claim.getValue().toString());
+    }
+
+    private AbstractUserStoreManager getUserStoreManager(int tenantId) throws UserStoreException {
+        return (AbstractUserStoreManager) EKYCServiceDataHolder.getInstance().getRealmService()
+                .getTenantUserRealm(tenantId).getUserStoreManager();
+    }
+
+
+    private boolean isUpdatePossible(EKYCVerifiedCredentialDTO ekycVerifiedCredentialDTO, int tenantId) throws
+            UserStoreException {
+        return !getUserStoreManager(tenantId).isReadOnly() &&
+                ekycVerifiedCredentialDTO != null &&
+                ekycVerifiedCredentialDTO.getStatus().equals(UserEKYCConstants.EKYCProccessStatuses.FINISHED);
```

Comment:
```suggestion
                UserEKYCConstants.EKYCProccessStatuses.FINISHED.equals(ekycVerifiedCredentialDTO.getStatus());
```
If there is any chance for ekycVerifiedCredentialDTO.getStatus() to be null let's change like this or use StringUtils.equals

Reply:
Done